hcl2template/types.packer_config.go

```diff
@@ -128,7 +129,7 @@ func (cfg *PackerConfig) EvalContext(ctx BlockContext, variables map[string]cty.
 	// dependency tree, so that any block can use any block whatever the
 	// order.
 	switch ctx {
-	case LocalContext, BuildContext:
+	case LocalContext, BuildContext: // todo: refine
```

Comment:
Since we are doing 'smarter' things here, maybe we could completely remove these contexts ? Not sure if that would help.

Reply:
It would simplify it. Was there any other reason for these contexts to exist? I was kind of going for lowest-touch since it's not code I'm deeply familiar with, but you're right that it doesn't seem necessary any more.

Reply:
I'll simplify it some but not remove the switch entirely because NilContext still trips out of this case and I want to keep that (used with packer console)